components/local_date_time/local_date_time.tsx

```diff
@@ -48,14 +48,25 @@ class LocalDateTime extends React.PureComponent<Props> {
             titleString = momentDate.toString() + ' (' + momentDate.tz() + ')';
         }
 
-        const timezoneProps = enableTimezone && timeZone ? {timeZone} : {};
         const options = {
-            ...timezoneProps,
-            hour12: !useMilitaryTime,
+            hour: 'numeric',
+            minute: 'numeric',
         };
-        let formattedTime = this.props.intl.formatTime(date, options);
+        if (enableTimezone && timeZone) {
+            options.timeZone = timeZone;
+        }
+        if (useMilitaryTime) {
+            options.hourCycle = 'h23';
+        } else {
+            options.hourCycle = 'h12';
+        }
 
-        if (formattedTime === String(date)) {
+        let formattedTime;
+        try {
```

Comment:
I believe the previous check was effectively a mechanism to detect that `formatTime` was unsuccessful. `Intl.DateTimeFormat` happily throws an exception in this case, so I'm handling it that way.